erpnext/hr/doctype/employee/employee.py

```diff
@@ -68,8 +68,10 @@ def update_user_permissions(self):
 		if not self.create_user_permission: return
 		if not has_permission('User Permission', ptype='write'): return
 
-		add_user_permission("Employee", self.name, self.user_id)
-		set_user_permission_if_allowed("Company", self.company, self.user_id)
+		if not frappe.db.exists({'doctype':'User Permission','user':self.user_id, 'allow':'Company', 'for_value': self.company}):
+			set_user_permission_if_allowed("Company", self.company, self.user_id)
+		if not frappe.db.exists({'doctype':'User Permission','user':self.user_id, 'allow':'Employee', 'for_value': self.name}):
```

Comment:
This is already checked within `add_user_permission`. There was an error in `add_user_permission `, thats why it was creating duplicate. We have fixed that, also written a patch to remove the duplicate.